modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/WindowsAPIUtils.java

```diff
@@ -0,0 +1,154 @@
+            int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+            userStoreManager = realmService.getTenantUserRealm(tenantId).getUserStoreManager();
+        } catch (UserStoreException e) {
+            String msg = "Error occurred while retrieving current user store manager";
+            throw new MDMAPIException(msg, e);
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+        return userStoreManager;
+    }
+
+    public static NotificationManagementService getNotificationManagementService() {
+        NotificationManagementService nmService;
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext ctx = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        ctx.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        ctx.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+        nmService =
+                (NotificationManagementService) ctx.getOSGiService(NotificationManagementService.class, null);
+        PrivilegedCarbonContext.endTenantFlow();
+        return nmService;
+    }
+
+    public static MediaType getResponseMediaType(String acceptHeader) {
+        MediaType responseMediaType;
+        if (MediaType.WILDCARD.equals(acceptHeader)) {
+            responseMediaType = MediaType.APPLICATION_JSON_TYPE;
+        } else {
+            responseMediaType = MediaType.valueOf(acceptHeader);
+        }
+        return responseMediaType;
+    }
+
+    public static Response getOperationResponse(List<String> deviceIDs, Operation operation,
+                                                Message message, MediaType responseMediaType)
+            throws DeviceManagementException, OperationManagementException {
+        WindowsDeviceUtils deviceUtils = new WindowsDeviceUtils();
+        DeviceIDHolder deviceIDHolder = deviceUtils.validateDeviceIdentifiers(deviceIDs,
+                message, responseMediaType);
+        getDeviceManagementService().addOperation(operation, deviceIDHolder.getValidDeviceIDList());
+        if (!deviceIDHolder.getErrorDeviceIdList().isEmpty()) {
+            return javax.ws.rs.core.Response.status(Constants.StatusCodes.
+                    MULTI_STATUS_HTTP_CODE).type(
+                    responseMediaType).entity(deviceUtils.
+                    convertErrorMapIntoErrorMessage(deviceIDHolder.getErrorDeviceIdList())).build();
+        }
+        return javax.ws.rs.core.Response.status(javax.ws.rs.core.Response.Status.CREATED).
+                type(responseMediaType).build();
+    }
+
+    public static PolicyManagerService getPolicyManagerService() {
+        PolicyManagerService policyManager;
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext ctx = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        ctx.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        ctx.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
```

Comment:
Why are we setting SUPER_TENANT_DOMAIN_NAME and SUPER_TENANT_ID here?
Is this implementation only bound to super tenant?